.github/actions/run-integration-tests/action.yml

```diff
@@ -25,32 +19,28 @@ runs:
       uses: actions/download-artifact@v2
       with:
         name: kubectl-gadget-linux-amd64-tar-gz
-        path: /home/runner/work/inspektor-gadget/
+        path: /home/runner/work/inspektor-gadget/inspektor-gadget/
     - name: Integration tests
       shell: bash
       run: |
-        export CONTAINER_REPO=${{ inputs.container_repo }}
-        export IMAGE_TAG=${{ inputs.image_tag }}
-        echo "IntegrationTestsJob: Using IMAGE_TAG=${IMAGE_TAG}"
-
-        tar zxvf /home/runner/work/inspektor-gadget/kubectl-gadget-linux-amd64.tar.gz
-        mv kubectl-gadget kubectl-gadget-linux-amd64
+        tar zxvf /home/runner/work/inspektor-gadget/inspektor-gadget/kubectl-gadget-linux-amd64.tar.gz
 
-        cleanup() { \
-            echo "IntegrationTestsJob: Workflow run is being cancelled: $1 was received"; \
-            trap - $1; \
-            if [[ $1 == "SIGINT" ]]; then \
-              echo "IntegrationTestsJob: Start the clean-up..."; \
-            else \
-              echo "IntegrationTestsJob: Just wait until the clean-up finishes..."; \
-              return; \
-            fi; \
-            echo "IntegrationTestsJob: Notifying the integration tests container about the cancellation"; \
-            docker kill --signal="SIGINT" ig-integration-tests > /dev/null; \
-            echo "IntegrationTestsJob: Waiting for the integration tests container to finish"; \
-            docker wait ig-integration-tests & wait $!; \
-            echo "IntegrationTestsJob: We are done with the clean-up. Let the job exit"; \
-            exit 0; \
+        cleanup() {
+            echo "IntegrationTestsJob: Workflow run is being cancelled: $1 was received"
+            trap - $1
+            if [[ $1 == "SIGINT" ]]; then
+              echo "IntegrationTestsJob: Start the clean-up..."
+            else
+              echo "IntegrationTestsJob: Just wait until the clean-up finishes..."
+              return
+            fi
+            echo "IntegrationTestsJob: Notifying the integration tests process about the cancellation"
+            MAKE_PID=$(jobs -p)
+            kill -2 $(pidof integration.test) > /dev/null
+            echo "IntegrationTestsJob: Waiting for the integration tests process to finish"
+            wait $MAKE_PID
```

Comment:
Could you please explain why to kill `pidof integration.test` but wait for `jobs -p`?

Reply:
Thanks for spotting, I changed to `pidof integration.test` in all the places since that is more clear.

Reply:
Also, the cleanup function is [working fine](https://github.com/mqasimsarfraz/inspektor-gadget/actions/runs/3091476571/jobs/5002582050) without  `;\`

Reply:
If you tested the cleanup after you switched to use `pidof integration.test`, we are ready to go 🚀

Reply:
Found minor [issue](https://github.com/mqasimsarfraz/inspektor-gadget/actions/runs/3093036278/jobs/5005205344) with the approach. So I am going to [stick](https://github.com/mqasimsarfraz/inspektor-gadget/actions/runs/3093714084/jobs/5006572610) with what I had before, added comment why we need it. We are ready for merge if you think it is looking good! :partying_face:

Reply:
The comment is clear, so is enough for now. We can investigate later if killing and waiting for the same process is possible. It would make the code clear.